packages/angular/cli/src/command-builder/utilities/json-schema.ts

```diff
@@ -43,6 +43,54 @@ export interface Option extends yargs.Options {
    * If this is falsey, do not report this option.
    */
   userAnalytics?: string;
+
+  /**
+   * Type of the values in a key/value pair field.
+   */
+  itemValueType?: 'string';
+}
+
+function coerceToStringMap(dashedName: string, value: (string | undefined)[]) {
+  const stringMap: Record<string, string> = {};
+  for (const pair of value) {
+    // This happens when the flag isn't passed at all.
+    if (pair === undefined) {
+      continue;
+    }
+
+    const eqIdx = pair.indexOf('=');
+    if (eqIdx === -1) {
+      throw new Error(
```

Comment:
NIT:

There is a bug in Yargs' `coerce` function when an exception is thrown. According to the documentation (https://yargs.js.org/docs/#api-reference-coercekey-fn), the exception should be treated as a validation error, but it is currently being thrown as an unhandled error.

**Workaround:** You can address this by returning the error wrapped in a `Promise`:

```javascript
return Promise.reject(`Invalid value for argument: ${dashedName}, Given: '${pair}', Expected key=value pair`);
```

Alternatively, you can throw the error and mark the method as `async`:

```javascript
throw `Invalid value for argument: ${dashedName}, Given: '${pair}', Expected key=value pair`;
```

Reply:
Thanks! I thought I had tested this but I definitely saw this behavior with an earlier revision of this change. Doing the reject workaround.

Do you have an upstream issue reference for this? I only found an issue that was marked as fixed (https://github.com/yargs/yargs/issues/755).

Reply:
I'll try to see if I can find a quick way to repro but that's realistically independent of this change.